### ISSUE TITLE
Hazelcast MultiMap does not remove item with long key

### ISSUE THREAD

**Issue by iozcelik** [hazelcast/hazelcast-csharp-client#150]
I try to remove an item from my multi map but it does not work for long type keys. It is working with int and string types.

My code example:
```
var map = hzUnitOfWork.HazelcastClient.GetMultiMap<long,string>("testmulti");
map.Put(1,"a");
map.Put(1,"b");
map.Put(1,"a");
map.Put(1,"a");
var result1 = map.Get(1);
var isremoved = map.Remove(1,"a");
var result2 = map.Get(1);
```

If I change type to string or int from long, this code working expectedly.

**Comment by asimarslan** [comment]
Hi İsmail,

`multiMap.Remove` method has a key parameter with Object type. So you should pass an actual long value with an `l` letter at the end.

İf you replace the line as 
```
var isremoved = map.Remove(1L,"a");
```
or
```
long key=1;
var isremoved = map.Remove(key,"a");
``` 

A better fix is changing the `Remove` method but we cannot do it for backward compatibility.

**Comment by iozcelik** [comment]
Hi Asim,

Actually I used like;

long key=1;
var isremoved = map.Remove(key,"a");

And doesn't work. I try again. Because I used entity.Id key.
May be I guess nullable long not working.

**Comment by iozcelik** [comment]
Issue looking closing but today I get the error.

**Comment by sancar** [comment]
@iozcelik Can you reshare a full example that demonstrates the broken behaviour when you use `1L` instead of `1` or when you use `long key=1`;

I tried following and it succefully deletes
```
var config = new ClientConfig();
var client = HazelcastClient.NewHazelcastClient(config);

var map = client.GetMultiMap<long,string>("testmulti");
long key = 1;
map.Put(1L,"a");
var isremoved = map.Remove(key,"a");
Console.WriteLine(isremoved);
```

**Comment by iozcelik** [comment]
@sancar I got the point on this issue. I use generic classes on my side. So, inside that classes I select id and try to remove with that Id. But I use "var" keyword not "long". So I added new method to my generic class and solve this problem.

My remove method like that;
`bool Remove(TKey key,TValue value) `

As @asimarslan said on 16 Feb 2018:"A better fix is changing the Remove method but we cannot do it for backward compatibility."
I think so, the better solution is need to change "Remove" method.
